### DISCUSSION TITLE
Why does the value of yaw continuosly change?

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#647] [Q&A] [answered]
Hi all,

I use kalman estimator as stabilizer. And I do not use external sensors or loco systems(now).

I find that, when I monitored the value of yaw by cfclient, sometimes it would continuously change. Specifically, it will start with like 2 degrees, and after 10 seconds, it will increase to 4 degrees, and it seems never converge. However, sometimes, it will be just fine, like it will stay close to 0, which is correct. On the other hand, the value of roll and pitch are always right.

I wonder why this happens. Thank you, guys!

**Comment by krichardsson** [comment, marked as answer]
Hi, 
both the accelerometers and gyros provide information for roll and pitch which makes them stable. For the yaw there is only gyro information available and thus it is no absolute measurement of the yaw, only the integrated gyro data. The noise in the gyro will potentially add some drift.

If you have a positioning system that provides yaw, like the lighthouse system, the yaw will behave in a similar way to roll and pitch.

For positioning systems that can not measure the yaw, like the Loco system or if you do not have any positioning system (your case), the yaw will slowly go towards zero (by design) when the Crazyflie is sitting on the ground. The idea is that we want to make it easy to get started, and this works fine as long as the user places the CF facing the positive X direction. 
When you are looking at the yaw, this is one factor to keep in mind. 

> However, sometimes, it will be just fine, like it will stay close to 0, which is correct.

This is what I would expect

> Specifically, it will start with like 2 degrees, and after 10 seconds, it will increase to 4 degrees, and it seems never converge. 
   
This is surprising and I do not know what it might be. I would not expect this unless there is some sort of input to the estimator from a deck or by pushing pose data via the radio.
You might see sudden changes when the kalman estimator resets (if the position is 100 meters away, or velocity over 10 m/s), but then it goes back to 0, not 4 degrees.

**Reply by cc299792458** [reply]
I see, thank you for the detailed explanations!
